napalm_eos/eos.py

```diff
@@ -85,6 +85,9 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
 
         self.enablepwd = optional_args.get('enable_password', '')
 
+    def __del__(self):
```

Comment:
See https://github.com/napalm-automation/napalm-eos/pull/149#issuecomment-285502603

It's possible (through exceptions, poorly written scripts, etc) to exit the python process without freeing the session on the switch.  Since the session is treated as a lock by NAPALM, and by default only 5 configuration sessions can be saved at a time, we should try really really hard to ensure that we're discarding configuration sessions when exiting.

It looks like napalm-base tries to accomplish this with __entry__ and __exit__, but those blocks are only called inside a `with` block.